### ISSUE TITLE
How can I include multiple generic intents across different handlers?

### ISSUE THREAD

**Issue by talkingnews** [alexa/alexa-skills-kit-sdk-for-nodejs#341]
tl;dr version of this question is: how, in node8, can I dynamically rename the state of an intent handler and include a common block of intents in each handler?

Let me explain! I'm still a bit of a JS/node noob, but I can see that states can exist between different handler objects. eg, from the audioplayer example:
```
var stateHandlers = {
playModeIntentHandlers:    Alexa.CreateStateHandler(constants.states.PLAY_MODE, {spoken intents},
remoteControllerHandlers : Alexa.CreateStateHandler(constants.states.PLAY_MODE, {remote control intents})
```
and then, in a different file:

`var audioEventHandlers = Alexa.CreateStateHandler(constants.states.PLAY_MODE, {intents from device during playback}`

I have 8 states; in each one, most requests which aren't valid in that state (like "shuffle on" in the info menu) will be handled by "unhandled", just saying "sorry, you can't do that here".

However, that still leave about 12 intents in each state which will be identical to each other and necessary to cleanly pass to the correct handler; so that's an intent to set the right state and then `emitWithState` to the correct handler for each possible intent, plus `stop`, `cancel`, `newsession`, `System.ExceptionEncountered`, `SessionEndedRequest` and `help` (with help taking its speech var from the constants.help.current_state. ie: `speechOutput = constants.help.this.handler.state`)

So that's a total of 96 duplicate states! What a waste of space, goes against DRY, not to mention maintenance headache if I ever change the System.ExceptionEncountered logging (although, at the moment, I simply point it to a logging function I wrote and call it with loggers.sx.call(this) n the intent)

Therefore, I'm thinking of something along the lines of:
```
var stateHandlers = {
playModeIntentHandlers:    Alexa.CreateStateHandler(constants.states.PLAY_MODE, {spoken intents},
remoteControllerHandlers : Alexa.CreateStateHandler(constants.states.PLAY_MODE, {remote control intents}
allOtherHandlers :         Alexa.CreateStateHandler(constants.states.PLAY_MODE, {generic exceptions, unhandled, sessionend etc} << how do I include this?!
```

Does that make it clear what I'm after? There was a similar request from someone else on Uservoice before "the big crash" so it's lost now.

Anyway, is this possible?

**Comment by TianTXie** [comment]
One scenario that you can use the concept of state to solve is that for same intent you have different response based on context. For the intents that just have one response in all states, you can just put all of them in the a handler that has no state.

**Comment by talkingnews** [comment]
Hi @TianTXie - thanks, that's a really good idea, too. In the end I settled for a combination of your idea, and then a catchall unhandledIntent which is then processed in another file.

**Comment by TianTXie** [comment]
Thanks for your interest in Alexa SDK. I'm closing this issue for now. Please feel free to reopen it if you encounter further issues.
